mockgen/parse_test.go

```diff
@@ -143,3 +146,127 @@ func TestParseArrayWithConstLength(t *testing.T) {
 		}
 	}
 }
+
+func Test_filterInterfaces(t *testing.T) {
+	type args struct {
+		all       []*model.Interface
+		requested []string
+	}
+	tests := []struct {
+		name    string
+		args    args
+		want    []*model.Interface
+		wantErr bool
+	}{
+		{
+			name: "no filter (returns all interfaces)",
+			args: args{
+				all: []*model.Interface{
+					{
+						Name: "Foo",
+					},
+					{
+						Name: "Bar",
+					},
+				},
+				requested: []string{},
+			},
+			want: []*model.Interface{
+				{
+					Name: "Foo",
+				},
+				{
+					Name: "Bar",
+				},
+			},
+			wantErr: false,
+		},
+		{
+			name: "filter by Foo",
+			args: args{
+				all: []*model.Interface{
+					{
+						Name: "Foo",
+					},
+					{
+						Name: "Bar",
+					},
+				},
+				requested: []string{"Foo"},
+			},
+			want: []*model.Interface{
+				{
+					Name: "Foo",
+				},
+			},
+			wantErr: false,
+		},
+		{
+			name: "filter by Foo and Bar",
+			args: args{
+				all: []*model.Interface{
+					{
+						Name: "Foo",
+					},
+					{
+						Name: "Bar",
+					},
+				},
+				requested: []string{"Foo", "Bar"},
+			},
+			want: []*model.Interface{
+				{
+					Name: "Foo",
+				},
+				{
+					Name: "Bar",
+				},
+			},
+			wantErr: false,
+		},
+		{
+			name: "incorrect filter by Foo and Baz",
+			args: args{
+				all: []*model.Interface{
+					{
+						Name: "Foo",
+					},
+					{
+						Name: "Bar",
+					},
+				},
+				requested: []string{"Foo", "Baz"},
+			},
+			want:    nil,
+			wantErr: true,
+		},
+		{
+			name: "missing interface (Baz not found)",
+			args: args{
+				all: []*model.Interface{
+					{
+						Name: "Foo",
+					},
+					{
+						Name: "Bar",
+					},
+				},
+				requested: []string{"Baz"},
+			},
+			want:    nil,
+			wantErr: true,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			got, err := filterInterfaces(tt.args.all, tt.args.requested)
+			if (err != nil) != tt.wantErr {
```

Comment:
instead of wantErr being a bool, could we make this a string and compare the error contents?